lua_modules/dht22/dht22.lua

```diff
@@ -0,0 +1,101 @@
+-- ***************************************************************************
+-- DHT22 module for ESP8266 with nodeMCU
+--
+-- Written by Javier Yanez 
+-- but based on a script of Pigs Fly from ESP8266.com forum
+--
+-- MIT license, http://opensource.org/licenses/MIT
+-- ***************************************************************************
+
+local moduleName = ...
+local M = {}
+_G[moduleName] = M
+
+local humidity
+local temperature
+local checksum
+local checksumTest
+
+function M.read(pin)
+  humidity = 0
+  temperature = 0
+  checksum = 0
+
+  -- Use Markus Gritsch trick to speed up read/write on GPIO
+  gpio_read = gpio.read
+  gpio_write = gpio.write
+  
+  bitStream = {}
```

Comment:
Always try to use locals. If you omit `local` keyword it becomes global -- this is standard catch.
Also think in closure terms. Try to hide everything private. Say, a cosher module

```lua
local M
do
  -- private stuff
  local not4all = { ... }
  local f1 = function(...) ... end
  local f2 = function(x, y) not4all[x] = y end
  -- explicitly expose
  M = {
    f1 = f1,
  }
end
return M
```

This way you hide details, returning just what explicitly exposed.

NB: _All above is my imho_.